src/plugins/http.js

```diff
@@ -138,12 +138,32 @@ function unpatch (http) {
 module.exports = [
   {
     name: 'http',
-    patch,
+    patch: function (http, tracer, config) {
+      patch.call(this, http, 'request', tracer, config)
+      if (semver.satisfies(process.version, '>=8')) {
+        /**
+         * In newer Node versions references internal to modules, such as `http(s).get` calling `http(s).request`, do
+         * not use externally patched versions, which is why we need to also patch `get` here separately.
+         */
+        patch.call(this, http, 'get', tracer, config)
+      }
```

Comment:
@rochdev I tried to scour the Node release notes for when/how this changed by searching for things like ‘module’ and ‘binding’ but came up short. In any case, I tried to describe the behaviour I was able to deduce as well as possible, let me know if you’d like to amend this.

Reply:
I would probably rephrase it simpler. Something like "Patch `http.get` as well in Node >=8 since it doesn't call `http.request`"

Reply:
Oh but it does https://github.com/nodejs/node/blob/v10.10.0/lib/https.js#L284. It’s just that something changed in Node between v6 and v8 that somehow makes it so that other functions exported from a module (such as `get`) than the patched function (such as `request`) still refer to the original unpatched implementation. So `get` needs to explicitly be patched here to trace it.

I wasn’t able to find anything in the Node changelog, but also didn’t have a great idea of what to search for.

Reply:
Well, it does call the internal `request` yes, but it doesn't call `http.request`.